index.html

```diff
@@ -5099,6 +5125,31 @@ <h2 id="defining-a-valid-audioworkletprcoessor">
                 <code>"name"</code>.
               </li>
             </ol>
+            <p>
+              The return value of this method controls the lifetime of the
+              <a>AudioWorkletProcessor</a>'s associated <a>AudioWorkletNode</a>
+              by determining the existence of the node's <a href=
+              "#dfn-active-reference">active reference</a>. At the conclusion
+              of each call to the <code>process()</code> method, if the result
+              of applying <a href=
+              "https://tc39.github.io/ecma262/#sec-toboolean"><code>ToBoolean</code></a>
+              (described in [[!ECMASCRIPT]]) to the return value is equal to
+              <code>true</code>, then a single <a>active reference</a> is
+              retained to the node, affecting the node's <a href=
+              "#lifetime-AudioNode">lifetime</a>. Otherwise, any active
+              reference is removed and the node's lifetime depends only on
+              other references as they may exist.
+            </p>
+            <div class="note">
+              While returning <code>true</code> ensures that
+              <code>process()</code> will be called during subsequent rendering
+              quanta, returning <code>false</code> does not prevent future
+              calls to <code>process()</code>. Other references to the
+              corresponding <a>AudioWorkletNode</a> may exist and the node may
+              thus continue to function. Furthermore, these future calls may
+              return a value of <code>true</code>, reinstating the active
+              reference.
```

Comment:
I think your description is basically the same as mine, although somewhat more complicated. The spec already says that a node can be removed when all of its references go away, and the result from `process()` governs one of the three possible reference types to an AWN (JS, has-inputs and processor-is-active).

However you're making an additional assertion here, which is that the UA must check _immediately_ after a `process()` invocation whether all the references have gone away. This seems to conflict with the spirit of the statements in the Lifetime section that the rules are guidance for the GC of nodes rather than absolute stipulations.

So far I don't see why this simplifies things at all for implementors, because they cannot tell whether JS or active-input reference types exist for the corresponding AWN. So it is already the case that an AWP could return false, but have `process()` called again (because a JS reference still exists).